components/expenses/ExpenseForm.js

```diff
@@ -123,7 +123,7 @@ export const prepareExpenseForSubmit = expenseData => {
   const isGrant = expenseData.type === expenseTypes.GRANT;
   const payee =
     expenseData.payee?.isNewUser || expenseData.payee?.isInvite
-      ? pick(expenseData.payee, ['name', 'slug', 'email', 'legalName', 'organization', 'newsletterOptIn'])
+      ? pick(expenseData.payee, ['id', 'name', 'slug', 'email', 'legalName', 'organization', 'newsletterOptIn'])
```

Comment:
Perhaps we shouldn't use pick here, see next line how this can mean either legacyId or a gQLv2 id.

Reply:
I did that to stay close to the initial approach: https://github.com/opencollective/opencollective-frontend/pull/8740/files#diff-fb75990d7ea60a101a18c88911bef1c9c8c9c6cdd0455acbe25096cc35ab6f9aL223. Will ush a safer fix